modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -146,6 +152,111 @@ public void tearDown() throws Exception {
 		_bundleActivator.stop(_bundleContext);
 	}
 
+	@Test
+	public void testAddUserWithWorkflowForLDAPUserWithLDAPPasswordPolicy()
+		throws Exception {
+
+		PasswordPolicy passwordPolicy =
+			_passwordPolicyLocalService.getDefaultPasswordPolicy(
+				TestPropsValues.getCompanyId());
+
+		passwordPolicy.setChangeRequired(true);
+		passwordPolicy.setCheckSyntax(true);
+
+		passwordPolicy = _passwordPolicyLocalService.updatePasswordPolicy(
+			passwordPolicy);
+
+		try (SafeCloseable safeCloseable =
+				_setLDAPAuthConfigurationWithSafeCloseable(true)) {
+
+			User user = _createUser(true, "abc");
+
+			Assert.assertEquals(
+				"User was created with incorrect LDAP Server Id", 1,
+				user.getLdapServerId());
+			Assert.assertFalse(
+				"During creation, LDAP user had portal password policy applied",
+				user.isPasswordReset());
+			Assert.assertNull(
+				"LDAP user is not bypassing portal password policy",
+				user.getPasswordPolicy());
+		}
+		catch (Throwable throwable) {
+			if (throwable instanceof UserPasswordException) {
+				throw new Exception(
+					"Exception relate to password policy was thrown",
+					throwable);
+			}
+
+			throw new Exception(
+				"Exception unrelated to password policy was thrown", throwable);
+		}
+		finally {
+			passwordPolicy.setChangeRequired(false);
+			passwordPolicy.setCheckSyntax(false);
+
+			_passwordPolicyLocalService.updatePasswordPolicy(passwordPolicy);
+		}
+	}
+
+	@Test
+	public void testAddUserWithWorkflowForLDAPUserWithoutLDAPPasswordPolicy()
+		throws Exception {
+
+		PasswordPolicy passwordPolicy =
+			_passwordPolicyLocalService.getDefaultPasswordPolicy(
+				TestPropsValues.getCompanyId());
+
+		passwordPolicy.setChangeRequired(true);
+		passwordPolicy.setCheckSyntax(true);
+
+		passwordPolicy = _passwordPolicyLocalService.updatePasswordPolicy(
+			passwordPolicy);
+
+		try (SafeCloseable safeCloseable =
+				_setLDAPAuthConfigurationWithSafeCloseable(false)) {
+
+			_assertUserPasswordException(true, "abc");
+
+			_assertUserCreatedWithPasswordPolicy(true, "Liferay123");
+		}
+		finally {
+			passwordPolicy.setChangeRequired(false);
+			passwordPolicy.setCheckSyntax(false);
+
+			_passwordPolicyLocalService.updatePasswordPolicy(passwordPolicy);
+		}
```

Comment:
There is a lot of duplication in these tests. Why not introduce a new `_setDefaultPasswordPolicyWithSafeCloseable` that handles the set/unset for all cases? This test would look like:
```
		try (SafeCloseable safeCloseable1 =
				_setLDAPAuthConfigurationWithSafeCloseable(false);
		       SafeCloseable safeCloseable2 =
				_setDefaultPasswordPolicyWithSafeCloseable(
                                        passwordPolicy -> {
                                                passwordPolicy.setChangeRequired(true);
                                                passwordPolicy.setCheckSyntax(true);
                                 })) {

			_assertUserPasswordException(true, "abc");

			_assertUserCreatedWithPasswordPolicy(true, "Liferay123");
		}
```

Reply:
Using a lambda expression here is not something I can figure out.  I can pass in the `passwordPolicy` object as the argument, but I'm not sure how to achieve what you are going for here.